File: ruby/Problem07.rb
# Problem 7
# What is the 10 001st prime number?

n = 200000

# Succeeds in returning array of primes, but slow for large values of n
primes = (2..n).to_a.delete_if do |x|
  (2..Math.sqrt(x)).select {|i| x % i == 0} != []
end

#Returns the 10,001st element in the array primes

primes[10000]

File: python/Problem07.py
# Problem 7
# What is the 10 001st prime number?

def nth_prime(n):
    i = 2
    primes = []
    while(len(primes) < n):
       j = 2
       while(j <= (i/j)):
          if not(i%j): break
          j = j + 1
       if (j > i/j) : primes.append(i)
       i = i + 1
    return primes.pop()